src/App.tsx

```diff
@@ -1,60 +1,143 @@
 import './App.scss';
+import React, { useState } from 'react';
+// import React from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { FullTodo, Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+const prepareTodos = (
+  users: User[],
+  todos: Todo[],
+): FullTodo[] => {
+  return todos.map((todo) => ({
+    ...todo,
+    user: users.find((user) => user.id === todo.userId) || null,
+  }));
+};
+
+const todos = prepareTodos(usersFromServer, todosFromServer);
+const users = usersFromServer;
+
+export const App: React.FC = () => {
+  const [userFromSelect, setUser] = useState('0');
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [todosArr, setTodo] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserError(false);
+    const { value } = event.target;
+
+    setUser(value);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleError(false);
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleAddTodo = () => {
+    const currentUser = users
+      .find(user => user.name === userFromSelect) || null;
+
+    if (!currentUser) {
+      setUserError(true);
+    }
+
+    if (!title.length) {
+      setTitleError(true);
+    }
+
+    if (currentUser && title) {
+      const maxId = Math.max(...todos.map(todo => todo.id));
+      const todo: FullTodo = {
+        id: maxId + 1,
+        title,
+        completed: false,
+        userId: currentUser.id,
+        user: currentUser,
+      };
+
+      setTodo(current => ([
+        ...current,
+        todo,
+      ]));
+
+      setTitle('');
+      setUser('0');
```

Comment:
**suggestion**
i'd move this whole block into a separate function for readability

Reply:
moved part of this block, thanks